_studio/mfx_lib/decode/mpeg2/src/mfx_mpeg2_decode.cpp

```diff
@@ -19,3275 +19,1133 @@
-        m_vdPar.info.clip_info.width = par->mfx.FrameInfo.Width;
-        m_vdPar.info.clip_info.height = par->mfx.FrameInfo.Height;
-
-        m_vdPar.info.aspect_ratio_width = par->mfx.FrameInfo.AspectRatioW;
-        m_vdPar.info.aspect_ratio_height = par->mfx.FrameInfo.AspectRatioH;
-
-        mfxU32 frameRateD = par->mfx.FrameInfo.FrameRateExtD;
-        mfxU32 frameRateN = par->mfx.FrameInfo.FrameRateExtN;
-
-        if (0 != frameRateD && 0 != frameRateN)
-        {
-            m_vdPar.info.framerate = (double) frameRateN / frameRateD;
-            m_isFrameRateFromInit = true;
-        }
-
-        m_vdPar.info.color_format = GetUmcColorFormat((mfxU8)par->mfx.FrameInfo.ChromaFormat);
-
-        if (UMC::NONE == m_vdPar.info.color_format)
-        {
-            return MFX_ERR_INVALID_VIDEO_PARAM;
-        }
-
-        m_vdPar.lFlags |= par->mfx.DecodedOrder ? 0 : UMC::FLAG_VDEC_REORDER;
-        m_vdPar.lFlags |= UMC::FLAG_VDEC_EXTERNAL_SURFACE_USE;
-
-        switch(par->mfx.TimeStampCalc)
-        {
-            case MFX_TIMESTAMPCALC_TELECINE:
-                m_vdPar.lFlags |= UMC::FLAG_VDEC_TELECINE_PTS;
-                break;
-
-            case MFX_TIMESTAMPCALC_UNKNOWN:
-                break;
-
-            default:
-                return MFX_ERR_INVALID_VIDEO_PARAM;
-        }
-
-        m_vdPar.numThreads = (mfxU16)(par->AsyncDepth ? par->AsyncDepth : m_pCore->GetAutoAsyncDepth());
-
-        if (m_InitH || m_InitW)
-        {
-            if (2048 < par->mfx.FrameInfo.Width || 2048 < par->mfx.FrameInfo.Height)
-            {
-                return MFX_ERR_INVALID_VIDEO_PARAM;
-            }
-
-            if (m_InitW < par->mfx.FrameInfo.Width || m_InitH < par->mfx.FrameInfo.Height)
-            {
-                return MFX_ERR_INCOMPATIBLE_VIDEO_PARAM;
-            }
-        }
-
-        m_isDecodedOrder = (1 == par->mfx.DecodedOrder) ? true : false;
-    }
-    else
-    {
-        m_InitW = 0;
-        m_InitH = 0;
-    }
-
-    if (m_implUmc.get())
-        m_implUmc->Reset();
-
-    return MFX_ERR_NONE;
-}
-
-mfxStatus VideoDECODEMPEG2InternalBase::Close()
-{
-    m_implUmc->Close();
-
-    for (int i = 0; i < DPB; i++)
-        if(mid[i] >= 0)
-            m_FrameAllocator->DecreaseReference(mid[i]);
-
-    for (int i = 0; i < DPB; i++)
-    {
-        mid[i] = -1;
-    }
-
-    for (int i = 0; i < NUM_FRAMES; i++)
-    {
-        if(m_frame[i].Data) delete []m_frame[i].Data;
-        m_frame[i].Data = NULL;
-    }
-
-    if(m_FrameAllocator)
-        m_FrameAllocator->Close();
-
-    delete m_FrameAllocator;
-    m_FrameAllocator = 0;
-
-    mfxStatus mfxRes = MFX_ERR_NONE;
-
-    if (0 < allocResponse.NumFrameActual)
-    {
-        mfxRes = m_pCore->FreeFrames(&allocResponse);
-        memset(&allocResponse, 0, sizeof(mfxFrameAllocResponse));
-    }
-
-    if (0 < m_opaqueResponse.NumFrameActual && true == m_isOpaqueMemory)
-    {
-        mfxRes = m_pCore->FreeFra
```

Comment:
what actually does flag "force"?, I haven't fount any check for it

Reply:
out of date